var/spack/repos/builtin/packages/gcc/detection_test.yaml

```diff
@@ -1,12 +1,27 @@
 paths:
-  # Ubuntu 18.04, system compilers without Fortran
+  # Ubuntu 20.04, system compilers without Fortran. This
+  # test also covers which flags are expected to be used
+  # during the detection of gcc.
   - layout:
       - executables:
         - "bin/gcc"
         - "bin/g++"
-        script: "echo 7.5.0"
+        script: |
+          if [[ "$1" == "-dumpversion" ]] ; then
```

Comment:
Can you copy and paste into shellcheck with `#!/bin/sh` to get rid of bash? I think we're better off with `[ "$1" = -dumpversion ];` etc.

Reply:
https://github.com/spack/spack/pull/43613/commits/4ab5ef09c65e2460d7b6e4d8d1babfb59fba1572

Reply:
I'll submit later another PR to change other detection tests too.